workflows/energyplus.py

```diff
@@ -203,281 +190,306 @@ def run_energyplus(self, isIP, run_directory, file_name, args, by_api):
+                ColumnNames.Warnings: '',
+                ColumnNames.Runtime: 0,
+                ColumnNames.Version: current_version
+            }
+            return EPLaunchWorkflowResponse1(
+                success=False,
+                message="Incorrect Version found {}: {}!".format(current_version, file_name),
+                column_data=column_data
+            )
 
-            # if not isIP:  # if using SI output units just use readvars CLI option
-            #     command_line_args += ['--readvars']
+        if by_api:
+            # if calling through API, then the arguments don't include the E+ binary
+            command_line_args = []
+        else:
+            command_line_args = [energyplus_binary]
 
-            # add some config parameters
-            command_line_args += ['--output-prefix', file_name_no_ext, '--output-suffix', 'C']
+        if extension == '.imf':
+            command_line_args += ['--epmacro']
 
-            # add in simulation control args
-            if 'weather' in args and args['weather']:
-                command_line_args += ['--weather', args['weather']]
-            else:
-                command_line_args += ['--design-day']
-
-            # and at the very end, add the file to run
-            command_line_args += [full_file_path]
-
-            # run E+
-            if by_api:
-                # if by API then find the API wrapper relative to this workflow, import it, set up a callback, and run
-                eplus_dir = Path(__file__).parent.parent.absolute()
-                sys.path.insert(0, str(eplus_dir))
-                from pyenergyplus.api import EnergyPlusAPI
-                x = lambda msg: self.callback("API Callback: " + str(msg))
-                api = EnergyPlusAPI()
-                state = api.state_manager.new_state()
-                api.runtime.callback_message(state, x)
-                api.runtime.set_console_output_status(state, False)
-                cur_dir = os.getcwd()
-                os.chdir(run_directory)
-                eplus_return = api.runtime.run_energyplus(state, command_line_args)
-                os.chdir(cur_dir)
-                if eplus_return != 0:
-                    self.callback("E+ FAILED")
-                    return EPLaunchWorkflowResponse1(
-                        success=False,
-                        message="EnergyPlus failed for file: %s!" % full_file_path,
-                        column_data={}
-                    )
+        # So...ExpandObjects is weird in E+.  It doesn't like running things in directories and accidentally symlinks
+        # things into the current working directory even if you specify an output directory
+        # We're just going to run it ourselves.  (Side note it could be a similar problem for EPMacro)
+        if extension != '.epJSON':
+            if platform.system() == 'Windows':
+                expand_objects = os.path.join(energyplus_root_folder, 'ExpandObjects.exe')
```

Comment:
Now here is where things got a bit hairy.  We were previously using the EnergyPlus CLI to run ExpandObjects for us.  However, that functionality doesn't work with multiple files running in the same directory, because it puts temporary files in the current directory, namely an `in.idf` file.  These stomp on each other and cause it to fail.  Unfortunately, without a change to E+ code itself, there wasn't a quick solution except to just call ExpandObjects ourselves here, which isn't a big deal really.